firefox-ios/Client/Frontend/Browser/Tabs/Views/TabTrayViewController.swift

```diff
@@ -812,11 +888,134 @@ class TabTrayViewController: UIViewController,
     func didSelectSection(panelType: TabTrayPanelType) {
         guard tabTrayState.selectedPanel != panelType else { return }
 
-        setupOpenPanel(panelType: panelType)
+        if isTabTrayUIExperimentsEnabled {
+            let targetIndex = TabTrayPanelType.getExperimentConvert(index: panelType.rawValue).rawValue
+            guard let targetVC = childPanelControllers[safe: targetIndex],
+                  let currentVC = pageViewController?.viewControllers?.first as? UINavigationController,
+                  let currentIndex = childPanelControllers.firstIndex(of: currentVC)
+            else { return }
+
+            let reduceMotionEnabled = UIAccessibility.isReduceMotionEnabled
+            if !reduceMotionEnabled {
+                animateThemeTransition(fromIndex: currentIndex, toIndex: targetIndex)
+            }
+
+            let direction: UIPageViewController.NavigationDirection = targetIndex > currentIndex ? .forward : .reverse
+            pageViewController?.setViewControllers([targetVC],
+                                                   direction: direction,
+                                                   animated: !reduceMotionEnabled,
+                                                   completion: nil)
+        }
 
+        setupOpenPanel(panelType: panelType)
         let action = TabTrayAction(panelType: panelType,
                                    windowUUID: windowUUID,
                                    actionType: TabTrayActionType.changePanel)
         store.dispatch(action)
     }
+
+    // MARK: - UIPageViewControllerDataSource & UIPageViewControllerDelegate
+
+    func pageViewController(_ pageViewController: UIPageViewController,
+                            viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        guard let viewController = viewController as? UINavigationController,
+              let index = childPanelControllers.firstIndex(of: viewController),
+              index > 0 else { return nil }
+        return childPanelControllers[index - 1]
```

Comment:
Nit: what about something like this, so we don't check the index and avoid crashes 
```suggestion
        guard let viewController = ...,
                   let index = ...
        return childPanelControllers[safe: index - 1]
```